app_legacy/Helpers/render/game.php

```diff
@@ -385,6 +385,18 @@ function RenderRecentGamePlayers(array $recentPlayerData): void
             $userName,
             $activity
         );
+
+        // Check if $activity contains a message about an "Unknown macro", and
+        // if so, strip the RP and replace it with an outdated emulator warning.
```

Comment:
Note that 'Unknown macro' could also indicate an actual problem with the script, though in that case all users should be affected.

Reply:
Noted, though this does make me wonder if we should even explicitly state the user is using an outdated emulator, as the message could be erroneous.

Two potential ideas that may be better:

1. Show "⚠️ Playing $gameTitle", and wrap this content in a <span> with a `title` attribute saying there was a problem executing the RP script, likely caused by an outdated emulator.

2. Show "⚠️ Playing $gameTitle (There was a problem generating rich presence)"

Or maybe a mix of these two approaches. What do you think?

Reply:
What about just putting the broken string in the tooltip?

 "⚠️ Playing $gameTitle" with "🚩1-1 🚶[Unknown macro]Number(0xh4ed) 💰[Unknown macro]Number(0xh62b)" in the tooltip?

That way the bad text doesn't affect the layout, but it's still visible for diagnosing.

Reply:
I think that's a great idea. I've made this change in https://github.com/RetroAchievements/RAWeb/pull/1482/commits/08d46fee9492458cb9d382dcaae9702313eaeae4.